### PR TITLE
Additional schema for addon config

### PR DESCRIPTION
**Pull request by sogaani**
* Support [definitions and references](https://github.com/mozilla-services/react-jsonschema-form#schema-definitions-and-references).
* Support [property dependencies](https://github.com/mozilla-services/react-jsonschema-form#property-dependencies).

image

image

### CODE REVIEW

File: static/js/addons-config/object-field.js [hunk 1]
```diff
@@ -40,36 +43,69 @@ function ObjectField(
...
+
+ObjectField.prototype.onPropertyChange = function(name, field) {
   return function(value) {
+    const schema = this.schema;
     const newFormData = {};
     newFormData[name] = value;
 
     this.formData = Object.assign(this.formData, newFormData);
 
+    // modify port of form based on form data.
```
Comment (mrstegeman) [inline static/js/addons-config/object-field.js]: s/port/part/

File: static/js/addons-config/schema-utils.js [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 'use strict';
 
-/* globals SchemaUtils */
+/* globals SchemaUtils Validator */
```
Comment (mrstegeman) [inline static/js/addons-config/schema-utils.js]: Add comma after SchemaUtils

**Comment by sogaani** [comment]
@mrstegeman thank you for reviewing. I fixed typo.

**Comment by mrstegeman** [comment]
@sogaani Travis CI found some eslint issues:

```
/home/travis/build/mozilla-iot/gateway/static/js/addons-config/object-field.js
  51:49  error  Missing space before opening brace  space-before-blocks
  54:29  error  Missing space before opening brace  space-before-blocks
  56:33  error  Missing space before opening brace  space-before-blocks
  79:47  error  Missing space before opening brace  space-before-blocks
  83:72  error  Missing space before opening brace  space-before-blocks
```

**Comment by mrstegeman** [comment]
@hobinjk Any issues with this? Seems to work fine for me.

**Pull request merged**